sympy/stats/drv.py

```diff
@@ -166,7 +166,7 @@ def compute_characteristic_function(self, expr, **kwargs):
         else:
             raise NotImplementedError()
 
-    def restricted_domain(self, condition):
```

Comment:
what was `restricted_domain` doing?

Reply:
`restricted_domain` was returning the domain where the condition for which the probability is to be calculated holds. 
The stats documentation mentions that the function `where` should  return the domain where the condition is true, but a `NotImplementedError` was raised for discrete random variables. This modification fixes the error, and the result returned by `restricted_domain` can be obtained by using `where(condition).set`